sdcm/remote/libssh2_client/__init__.py

```diff
@@ -635,7 +635,8 @@ def _complete_run(self, channel: Channel, exception: Exception,  # pylint: disab
         result.stderr = stderr.getvalue()
         if channel is not None:
             try:
-                self.session.eagain(channel.close, timeout=self.timings.channel_close_timeout)
+                with self.session.lock:
```

Comment:
@dkropachev what do you think about this one ?

Reply:
also take a look at the commit message for explanation